Task 2: Data Cleaning & Preprocessing
By: Shahin
Dataset: Titanic (Kaggle)
Goal: Clean and prepare the dataset for analysis

In [2]:
import pandas as pd
import numpy as np

print(" Libraries imported successfully!")

 Libraries imported successfully!


In [3]:
titanic = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')
print(" Dataset loaded successfully!")
print("First look at data:")
print(titanic.head())

 Dataset loaded successfully!
First look at data:
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4    

In [4]:
print("Dataset Size:", titanic.shape)
print("\nColumn Names:", titanic.columns.tolist())
print("\nData Types:")
print(titanic.dtypes)
print("\nBasic Info:")
titanic.info()

Dataset Size: (891, 12)

Column Names: ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']

Data Types:
PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

Basic Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8 

In [5]:
print("Missing Values in Each Column:")
print(titanic.isnull().sum())
print("\nMissing Value Percentage:")
print((titanic.isnull().sum() / len(titanic)) * 100)

Missing Values in Each Column:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Missing Value Percentage:
PassengerId     0.000000
Survived        0.000000
Pclass          0.000000
Name            0.000000
Sex             0.000000
Age            19.865320
SibSp           0.000000
Parch           0.000000
Ticket          0.000000
Fare            0.000000
Cabin          77.104377
Embarked        0.224467
dtype: float64


In [14]:
titanic['Age'] = titanic['Age'].fillna(titanic['Age'].median())
titanic['Embarked'] = titanic['Embarked'].fillna(titanic['Embarked'].mode()[0])
if 'Cabin' in titanic.columns:
    titanic.drop(columns=['Cabin'], inplace=True)
    print("Cabin column dropped!")
else:
    print("Cabin already removed!")
print("\nMissing values after fixing:")
print(titanic.isnull().sum())
print("\n All missing values handled!")

Cabin already removed!

Missing values after fixing:
PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

 All missing values handled!


In [13]:
print("Duplicates before:", titanic.duplicated().sum())
titanic.drop_duplicates(inplace=True)
print("Duplicates after:", titanic.duplicated().sum())
print(" Duplicate check done!")

Duplicates before: 0
Duplicates after: 0
 Duplicate check done!


In [15]:
titanic['Sex'] = titanic['Sex'].map({'male': 0, 'female': 1})
titanic['Embarked'] = titanic['Embarked'].map({'S': 0, 'C': 1, 'Q': 2})
print("✅ Text converted to numbers!")
print(titanic[['Sex', 'Embarked']].head(10))

✅ Text converted to numbers!
   Sex  Embarked
0    0         0
1    1         1
2    1         0
3    1         0
4    0         0
5    0         2
6    0         0
7    0         0
8    1         0
9    1         1


In [16]:
titanic.drop(columns=['Name', 'Ticket', 'PassengerId'], inplace=True)
print("Unnecessary columns removed!")
print("Remaining columns:", titanic.columns.tolist())
print("Final shape:", titanic.shape)
print("\nFinal cleaned data:")
print(titanic.head())

Unnecessary columns removed!
Remaining columns: ['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Embarked']
Final shape: (891, 8)

Final cleaned data:
   Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    0  22.0      1      0   7.2500         0
1         1       1    1  38.0      1      0  71.2833         1
2         1       3    1  26.0      0      0   7.9250         0
3         1       1    1  35.0      1      0  53.1000         0
4         0       3    0  35.0      0      0   8.0500         0


In [17]:
titanic.to_csv('cleaned_titanic.csv', index=False)
print("Cleaned dataset saved!")
print("\nSummary of what I did:")
print("1. Loaded dataset - 891 rows, 12 columns")
print("2. Found missing values - Age, Cabin, Embarked")
print("3. Fixed missing values - Filled Age & Embarked, Dropped Cabin")
print("4. Checked duplicates - None found")
print("5. Converted text to numbers - Sex and Embarked")
print("6. Removed useless columns - Name, Ticket, PassengerId")
print("7. Final dataset - 891 rows, 8 columns")
print("\nTASK 2 COMPLETED!")

Cleaned dataset saved!

Summary of what I did:
1. Loaded dataset - 891 rows, 12 columns
2. Found missing values - Age, Cabin, Embarked
3. Fixed missing values - Filled Age & Embarked, Dropped Cabin
4. Checked duplicates - None found
5. Converted text to numbers - Sex and Embarked
6. Removed useless columns - Name, Ticket, PassengerId
7. Final dataset - 891 rows, 8 columns

TASK 2 COMPLETED!


My Conclusions from Data Cleaning:

1. Cabin column had too many missing values (77%) so I dropped it
2. Age was filled with median to avoid effect of outliers
3. Embarked had only 2 missing values so mode worked fine
4. Converting text to numbers is important for ML models
5. Clean data is the foundation of good analysis